kubetest/e2e.go

```diff
@@ -230,9 +231,17 @@ func run(deploy deployer, o options) error {
 							return federationTest(testArgs)
 						}))
 					} else {
-						errs = util.AppendError(errs, control.XmlWrap(&suite, "Test", func() error {
-							return test(testArgs)
-						}))
+						var tester Tester
+						tester = &GinkgoScriptTester{}
+						if testBuilder, ok := deploy.(testBuilder); ok {
+							tester, err = testBuilder.BuildTester(&o)
+							errs = util.AppendError(errs, err)
+						}
+						if tester != nil {
```

Comment:
we'd probably always fallback to `GinkgoScriptTester`

Reply:
So we do, unless BuildTester returns nil - we then assume that running the ginkgo tests is not appropriate.  `BuildTester` could always itself return the ginkgo tester in the case of error, but I think the right thing to do here is to fail the test.

Another way to say this: the fallback is for configurations that haven't implemented BuildTester, not for configurations where trying to build a tester object encounters an error.